src/providers/postgres/qgspostgresconn.cpp

```diff
@@ -251,10 +251,20 @@ QgsPostgresConn::QgsPostgresConn( const QString &conninfo, bool readOnly, bool s
   };
   addDefaultTimeoutAndClientEncoding( expandedConnectionInfo );
 
+  QgsDataSourceUri expandedUri( expandedConnectionInfo );
+  static QString lastUser;
```

Comment:
why is this static?

Reply:
@nyalldawson We need to remember the user from different Postgresconnections. The user configures a Postgres connection and loads a saved a map is from this connection. This map contains different authids, but the Postgres username is the same, so reusing the saved name works. So we need to store the name outside Postgres connection instances.

A class member variable is an alternative for the static, but I preferred the static because it is only used at this particular spot.

Reply:
> so threading aside it's not a safe assumption that one instance of QGIS should use one postgres username

Yes, but only when we don't know the username, we try to use to last successfully used name (just like PGUSER supplies a global default). Do you have a suggestion where/how to store it?